src/lib/AnalysisApps/Results/Utils/workflowApi.test.ts

```diff
@@ -0,0 +1,504 @@
+import {
+  GEN3_WORKFLOW_API,
+  getPresignedUrl,
+  useGetWorkflowDetailsQuery,
+  useGetWorkflowsQuery,
+  useGetWorkflowsMonthlyQuery,
+  useGetPresignedUrlOrDataForWorkflowArtifactQuery,
+} from './workflowApi';
+import { GEN3_API } from '@gen3/core';
+
+const GEN3_FENCE_API = `${GEN3_API}/user`; // TODO replace with GEN3_FENCE_API from gff core when updated
+
+const server = setupServer();
+
+import {
+  FetchArgs,
+  FetchBaseQueryError,
+  FetchBaseQueryMeta,
+  QueryReturnValue,
+} from '@reduxjs/toolkit/query';
+import {
+  describe,
+  expect,
+  it,
+  afterEach,
+  jest,
+  beforeAll,
+  beforeEach,
+  afterAll,
+} from '@jest/globals';
+
+type MaybePromise<T> = T | Promise<T>;
+
+type FetchMockResponse = { url: string };
+
+type RTKQFetchFunction = (
+  arg: string | FetchArgs,
+) => MaybePromise<
+  QueryReturnValue<FetchMockResponse, FetchBaseQueryError, FetchBaseQueryMeta>
+>;
+
+describe('getPresignedUrl', () => {
+  const fetchWithBQMock = jest.fn<RTKQFetchFunction>();
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return a presigned URL when the API call is successful', async () => {
+    const mockUid = '12345';
+    const mockMethod = 'download';
+    const mockUrl = 'https://example.com/presigned-url';
+    fetchWithBQMock.mockReturnValue({
+      data: { url: mockUrl },
+    });
+
+    const result = await getPresignedUrl(mockUid, fetchWithBQMock, mockMethod);
+
+    expect(fetchWithBQMock).toHaveBeenCalledWith({
+      url: `${GEN3_FENCE_API}/data/${mockMethod}/${mockUid}`,
+    });
+    expect(result).toEqual({ data: { url: mockUrl } });
+  });
+
+  it('should return an error when the API call fails', async () => {
+    const mockUid = '12345';
+    fetchWithBQMock.mockReturnValue({
+      error: { status: 500, data: undefined },
+    });
+
+    const result = await getPresignedUrl(mockUid, fetchWithBQMock);
+
+    expect(fetchWithBQMock).toHaveBeenCalledWith({
+      url: `${GEN3_FENCE_API}/data/download/${mockUid}`,
+    });
+    expect(result).toEqual({ error: { status: 500, data: undefined } });
+  });
+
+  it('should default to the "download" method if no method is provided', async () => {
+    const mockUid = '12345';
+    const mockUrl = 'https://example.com/presigned-url';
+    fetchWithBQMock.mockReturnValue({
+      data: { url: mockUrl },
+    });
+
+    await getPresignedUrl(mockUid, fetchWithBQMock);
+
+    expect(fetchWithBQMock).toHaveBeenCalledWith({
+      url: `${GEN3_FENCE_API}/data/download/${mockUid}`,
+    });
+  });
+});
+
+describe('workflowApi', () => {
+  beforeAll(() => {
+    // Start the interception.
+    server.listen();
+  });
+  beforeEach(() => {});
+
+  afterEach(() => {
+    // Remove any handlers you may have added
+    // in individual tests (runtime handlers).
+    server.resetHandlers();
+  });
+
+  afterAll(() => {
+    // Disable request interception and clean up.
+    server.close();
+  });
+
+  const detailsData = {
```

Comment:
Consider importing the test data from 

src/lib/AnalysisApps/Results/Utils/workflowApi.test.ts